deeprank/learn/DataSet.py

```diff
@@ -527,44 +490,75 @@ def _select_pdb(self, mol_names):
+        """
+
+        return_string = subject_string
+        for operator_string in ['>=', '<=', '==', '!=', '>', '<']:
+            search_index = 0
+            while search_index < len(return_string):
+                found_index = return_string.find(operator_string, search_index)
+
+                if found_index >= 0:
+                    # ATTENTION: we don't want to insert it twice!
+                    if return_string[:found_index].endswith(inserting_string):
+
+                        search_index = found_index + len(operator_string)
+                    else:
+                        return_string = return_string[:found_index] + inserting_string + return_string[found_index:]
+
+                        search_index = found_index + len(inserting_string) + len(operator_string)
+                else:
+                    # not found
+                    break
+
+        return return_string
+
     def filter(self, molgrp):
         """Filter the molecule according to a dictionary, e.g.,
-        dict_filter={'DOCKQ':'>0.1', 'IRMSD':'<=4 or >10'}).
+        dict_filter={'Pathogenic':'==1.0'}
 
         The filter is based on the attribute self.dict_filter
         that must be either of the form: { 'name': cond } or None
 
         Args:
-            molgrp (str): group name of the molecule in the hdf5 file
+            molgrp (hdf5 group): group of the molecule in the hdf5 file
         Returns:
             bool: True if we keep the complex False otherwise
 
         Raises:
             ValueError: If an unsuported condition is provided
         """
+
+        logger.debug("filtering {}".format(molgrp.name))
+
         if self.dict_filter is None:
             return True
 
         for cond_name, cond_vals in self.dict_filter.items():
 
-            try:
-                val = molgrp['targets/' + cond_name][()]
-            except KeyError:
-                warnings.warn(f'Filter {cond_name} not found for mol '
-                              f'{molgrp.name}')
+            logger.debug("filter with condition {}: {}".format(cond_name, cond_vals))
+
+            if cond_name not in molgrp['targets']:
+                raise ValueError(f'Filter {cond_name} not found for mol {molgrp.name}')
+
+            val = molgrp['targets/' + cond_name][()]
 
             # if we have a string it's more complicated
             if isinstance(cond_vals, str):
-                ops = ['>', '<', '==', '<=', '>=']
-                new_cond_vals = cond_vals
-                for o in ops:
-                    new_cond_vals = new_cond_vals.replace(
-                        o, 'val' + o)
+                new_cond_vals = DataSet._insert_before_operators(cond_vals, 'val')
                 if not eval(new_cond_vals):
                     return False
             else:
-                raise ValueError(
-                    'Conditions not supported', cond_vals)
+                raise ValueError("Conditions not supported", cond_vals)
 
         return True
 
```

Comment:
Line 584 chain_tags can be removed.
Lines 629:632  need to be changed.

Reply:
I made a change here: https://github.com/DeepRank/deeprank/commit/fe5422da5595400b32237d9e6633f5d4fee48e74

Reply:
Thanks Coos